src/Nethermind/Nethermind.Blockchain/Find/LogFinder.cs

```diff
@@ -66,12 +69,22 @@ private bool ShouldUseBloomDatabase(BlockHeader fromBlock, BlockHeader toBlock)
 
         private IEnumerable<FilterLog> FilterLogsWithBloomsIndex(LogFilter filter, BlockHeader fromBlock, BlockHeader toBlock)
         {
+            Block FindBlock(long blockNumber)
```

Comment:
did you test it with cases where you have multiple blocks on the level?

Reply:
it will fail miserably there, need to handle that case where you have multiple blocks on level

Reply:
I want canonical block, not an issue.

Reply:
I though it was on migration, not on RPC call, here it is less severe